### PR TITLE
[IMP] website: improve perf: use module info from cache instead of ref

### PR DESCRIPTION
**Pull request by Gorash**
This improvement remove 12 requests on all page when the user is connected with published access.

### CODE REVIEW

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Hello :wave: 

> This improvement remove 13 request on all page when the user is connected.

That's not true, right? It only impact the admin/designer of the DB, not all connected user (no change for internal users and portal users).

I am surprised there is no perf test failing, since it indeed reduce SQL queries for admin/designer, could you add a quick one? That would be cool to have a test and unsure this kind of improvement is not reverted in x months :)

Thank you !

File: odoo/addons/base/models/ir_module.py [hunk 1]
```diff
@@ -947,6 +947,11 @@ def _get_id(self, name):
         self.env.cr.execute("SELECT id FROM ir_module_module WHERE name=%s", (name,))
         return self.env.cr.fetchone()
 
+    @tools.ormcache('name')
+    def _get_data(self, name):
```
Comment (rdeodoo) [inline odoo/addons/base/models/ir_module.py:951]: Could you please name it like we already do for website and lang module so it is easier to grep and figure stuff?
See `_get_cached()` and `_get_cached_values()`

**Comment by Yenthe666** [comment]
I wish we could get stuff like this in older versions 😄

**Comment by Gorash** [comment]
@rdeodoo it seems to me that it is for all those connected since there is this `t-set` above. I haven't seen a `groups` containing this template part.
https://github.com/odoo/odoo/pull/83308/files#diff-1056cfb1cb521a54017d225630b404af931fbc1d14893a8b9d7f281443bfa429L117

(I used `_get_data` as there is `get_id` upper, but it is true that it will be clearer)

I will check, adapt the code and I will add tests ;)

**Comment by rdeodoo** [comment]
> it seems to me that it is for all those connected since there is this t-set above. I haven't seen a groups containing this template part.

There is, all those are contained inside this div :-)
```xml
<div class="o_menu_systray d-none d-md-flex" groups="website.group_website_publisher">
```

Still, this is a good imp to make I guess, even if it impacts only admin/publisher :+1:

File: addons/website/tests/test_performance.py [hunk 2]
```diff
@@ -101,6 +101,22 @@ def test_20_perf_sql_queries_homepage(self):
...
+    def test_21_perf_sql_queries_homepage(self):
+        # homepage loaded with a connected user
+        self.authenticate('demo', 'demo')
+        self.assertEqual(self._get_url_hot_query('/'), 21)
+
+    def test_22_perf_sql_queries_homepage(self):
+        # homepage loaded with a publisher access
+        self.authenticate('demo', 'demo')
+        self.env.ref('website.group_website_publisher').write({'users': [(4, self.env.ref('base.user_demo').id)]})
+        self.assertEqual(self._get_url_hot_query('/'), 24)
```
Comment (rdeodoo) [inline addons/website/tests/test_performance.py:113]:
```suggestion
        extra_companies_count = self.env['res.company'].search_count([]) - 1
        self.authenticate('demo', 'demo')
        self.env.ref('website.group_website_publisher').write({'users': [(4, self.env.ref('base.user_demo').id)]})
        self.assertEqual(self._get_url_hot_query('/'), 24 + extra_companies_count)
```

**Review by rdeodoo — COMMENTED** [review]
Runbot diff 24 v 25 comes from a second company being created when running the test, I guess from account module or something.

I suggested a diff to bypass that and keep the test working both on local with only website and on runbot.

**Comment by rdeodoo** [comment]
@Gorash Could you have a look at the commit I just pushed? I'd like to keep the same signature for `_get_cached_values()` and `_get_cached()` as we do for other model.
Eg: `_get_cached_values()` cached on the record rather than just a method that takes a name. (we can add other return value if needed one day)

What do you think?

**Review by rdeodoo — COMMENTED** [review]
After further discuss internally, as this is only improving perfs for users with edition rights (`website.group_website_publisher`), I am not so sure this is worth introducing a cache mechanism for this model (as we do for lang/website).

As:
- this cache methods most likely won't be of any use elsewhere (as opposed to the cache for website/lang which is used at multiple place)
- would just benefit to the editors users
- makes the code a bit harder to read

I don't think that's actually worth merging. Should you have another opinion or someone else thinking it is worth, let me know, I am willing to discuss it and change my mind here :+1:

**Comment by Gorash** [comment]
I understand. It was a proposal, I try to remove all the sql in the renderings so as not to have to put a higher level (t-)cache. For me, there is no problem closing if you find that it does not have enough benefit in relation to readability.

**Pull request closed without merging**